### PR TITLE
New Feature: read args from config file

### PR DESCRIPTION
**Pull request by andrewjhumphrey**
Allow the default of command line flags via the use of a config file.
It's pretty rudimentary, but it should work okay for now.

Example `.iamy-flags` file:

```
--skip-cfn-tagged --skip-tagged iamy-ignore
```

### CODE REVIEW

File: iamy.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ var (
 	defaultDir      string
 	dryRun          *bool
 	versionFileName string = ".iamy-version"
+	configFileName  string = ".iamy-config"
```
Comment (stevehodgkiss) [inline iamy.go]: I'm wondering if this should this be called `.iamy-flags`, to better suit it's contents:

```
❯ cat .iamy-config
--skip-tagged=iamy-ignore
```

Comment (andrewjhumphrey) [reply iamy.go]: Sure works for me.

**Comment by andrewjhumphrey** [comment]
Bummer, we have a bit of a design issue.

command line flags are specific to the command being run (ie pull takes --skip-tagged, but push doesn't).
So if you put --skip-tagged in the .iamy-flags file it complains about an unknown option being passed.

I see 3 ways to fix this:
1.  Make all flags available to all commands, they just get ignored by the command if they aren't applicable.
2. Make the flag parser skip unknown flags from the config file
3. Make a separate .iamy-flags  file for each command with specific flags. (ie .iamy-flags-pull and a "global" .iamy-flags) and read the correct files

I'm leaning towards option 1.
Anyone else got strong opinions?

**Comment by simpsora** [comment]
How about option 4, per-command flags in the single file?

```yml
pull:
  --skip-tagged
push:
  --foo
```

How far away from upstream are we, and do we intend to pull in any upstream changes?  Option 1 especially and option 2 will make that process a bit harder.  Unless there's good reason to do so, I reckon we should try to keep our local changes constrained to fewer places.

**Comment by andrewjhumphrey** [comment]
> How about option 4, per-command flags in the single file?
> 
> ```yaml
> pull:
>   --skip-tagged
> push:
>   --foo
> ```


Yep, we could do that.  It's a bucket load more effort, the command line parser already in use can just read from a file, it has no idea of structure.  We'd need to replace it with something else,  not that big a deal I guess.  But it would also require some upstream unfriendly changes.

There is very little, if any, action going on upstream,  but I do intend on pulling in or replicating interesting bits when and if they release them.

**Comment by andrewjhumphrey** [comment]
Actually thinking about it a little more I think we need to support the same --skip type flags for both push and pull.

A push first does a pull, under the covers, so it has something to compare to and a pull needs
And a pull needs to not pick up crap we don't care about (and clean it up if --delete is specified

File: README.md [hunk 2]
```diff
@@ -7,13 +7,24 @@ This allows you to use an [Infrastructure as Code](https://en.wikipedia.org/wiki
...
-The additional features we are likely to add to this version are:
-- .iamy-version file support, [Original PR](https://github.com/99designs/iamy/pull/63)
+The additional features we are likely to add to this fork are:
 - support for organizations, ous and scps
-- allow the skipping of s3 policy management
+
+# Installation
+
+brew tap envato/envato-iamy
+brew install envato/envato-iamy/iamy
```
Comment (patrobinson) [inline README.md:26]: Thank you

File: README.md [hunk 3]
```diff
@@ -71,10 +82,10 @@ Exec all aws commands? (y/N) y
 
 ## Accurate cloudformation matching
 
-By default, iamy will use a simple heuristic (does it end with an ID, eg -ABCDEF1234) to determine if a given resource is managed by cloudformation. 
+By default, iamy will use a simple heuristic (does it end with an ID, eg -ABCDEF1234) to determine if a given resource is managed by cloudformation.
 
 This behaviour is good enough for some cases, but if you want slower but more accurate matching pass `--accurate-cfn`
-to enumerate all cloudformation stacks and resources to determine exactly which resources are managed. 
+to enumerate all cloudformation stacks and resources to determine exactly which resources are managed.
```
Comment (patrobinson) [inline README.md:88]: we can probably remove this from the readme.
I'd like to dump that code all together eventually

**Review by patrobinson — APPROVED** [review]
:+1:

**Review by stevehodgkiss — APPROVED** [review]

**Pull request merged**